### ISSUE TITLE
Fix commit history

### ISSUE THREAD

**Issue by pro-src** [VeNoMouS/cloudscraper#23]
Ideally, if you want to flatten a github repo, you would rebase in order to squash. However, the branch was completely recreated from nothing and my commits were lost from your repository. I have the contributor status but I co-authored this project and there isn't any proof of that while the code that I only contributed remains.

https://github.com/pro-src/cloudscraper.py/tree/ce939982607c3059a94cb86e33ba61430c11f989

*I provided code on discord* that was then added in your dev branch and committed by you to your dev branch at least a few times as well. I don't care about that, although now it's not clear to a third party what my contributions were entirely. In the future, I would advise against it.

![github com_VeNoMouS_cloudscraper](https://user-images.githubusercontent.com/34285059/57169661-aa09c100-6dcd-11e9-8808-34eb23be80a2.png)
That hardly seems fair :point_up: (It should say 2 contributors)

I'm asking you to go back and rebase since this violates the MIT license and the spirit of Github based contributions aka I want credit where it is due. The license header doesn't matter since the Git history shows who authored which code (minus discord).

I know your commit messages were ugly, empty messages aside from a dot but you can fix that during an interactive rebase.

And lets not forget this comment that you've made: https://github.com/VeNoMouS/cloudscraper/issues/13#issuecomment-488169156

I wouldn't want you to be a hypocrite. Just my 2 cents...

**Comment by pro-src** [comment]
@pannal

**Comment by VeNoMouS** [comment]
Oh @pro-src sorry man, didnt mean to wipe you off there I was just trying to clean up the history a little bit

**Comment by VeNoMouS** [comment]
Get in discord I only just got home

**Comment by pro-src** [comment]
@VeNoMouS I don't mind helping or even doing the rebase myself (You could clone and force push it). The only thing that really bothers me about this is the current forks since they won't be able to easily pull or send valid PRs without re-forking/cloning the repo first. That said, I've thought about this and I think it should be done while there are only 3 such forks, only a small group of people will potentially be affected if it's done soon.

Sometime tomorrow (your time), I'll get on discord or if you want, I'll do the rebase myself.

**Comment by VeNoMouS** [comment]
Cheers mate :) I actually think its from when went "live" and if you recall i wiped out dev branch and did a clean push of all our work and removed all the mess that was in there while we initially worked on it...

**Comment by pannal** [comment]
Why exactly was I highlighted for this? :D

**Comment by pro-src** [comment]
@pannal I thought you might find it interesting. :innocent:

 I'm re-opening this since it still hasn't been fixed.

**Comment by vn-ki** [comment]
Please never rewrite git history. It is a very bad practice and you might end up creating headaches for any number of people who have cloned or forked the project.

**Comment by pro-src** [comment]
Hi @vn-ki thanks for your input. This something that should never ever usually be done to the master branch of a repository but, like most things, there is an exception.

At this early stage of this very new project, I only know of 4 such people that would be affected. You can check the network members: https://github.com/VeNoMouS/cloudscraper/network/members

So once this issue has been taken care of there is nothing further to worry about.

**Comment by pro-src** [comment]
On second thought, if simply keeping the other history in a separate branch will fix the repo stats, etc. then I'd be happy with that form of a historical record.

Although, I'm not sure if it will exactly... Obviously, I don't have control here and will take what I can get.

**Comment by pro-src** [comment]
@VeNoMouS It's been 3 days, I've been on discord multiple times, and provided step by step instructions on the rebase procedure. If you will kindly and simply just push a branch to repo so this can be closed without annoying anybody. Here is a simpler set of commands to do just that.

```sh
git remote add pro-src https://github.com/pro-src/cloudscraper.py
git fetch pro-src
git checkout -b v0.0.0 74bf9aa4951d4408451a9f6b1e01b8f2d93cc94c
git push origin v0.0.0
```

After that, if you want to edit commit messages, etc., feel free to do an interactive rebase.

~Thanks

**Comment by VeNoMouS** [comment]
Just so your all aware, I have had to block pro-src from my repo and discord due to comments directed at myself and other people that I feel don't belong in this public forum, after his outburst in discord, he has filed a `DMCA takedown notice` with github.

Anyway this is petty, and we will see what github responds with.

**Comment by miigotu** [comment]
This is actually a pretty serious issue. It seems minor to you which i sort of understand, but his DMCA is actually valid if you remove history and credits because it breaks the licensing and is actually illegal. Commit and contribution history, even if it looks messy, doesn't hurt the project whatsoever.

I'm not saying this to attack you or even place blame, I just don't think you understand how serious of an infraction this is for open source licenses. I know you did not intend this as something to harm other conttributors, but if you try and see it from the other side you would hopefully just want to fix the issue so there is no controversy and there is no perceived disrespect to contributors or the movement.

I wanted to use your project and have people in the SC community contribute back, but currently I am unable to do that with controversy and lack of trust. If you need help getting this back into a community-acceptable state and repair that trust I am 100% willing to help you do that just as I am sure it hers will. Good, useful code is only a part of the equation for a successful project. I have been maintaining or leading projects for nearly 20 years, so if you need any help or advice, or even some light mentoring (I'm not sure of your experience level as a maintainer) I am here any time, just shoot me a mail or mention.

I'm sure you could also ask the other serial maintainer on SC @WebSpider as well, as he believes in protecting the community and the project is paramount just like me.

**Comment by pannal** [comment]
- which license has been broken explicitly, when, how and by whom?
- how does an accidental force-push exactly infringe what, how and by whom?
- `I'm not saying this to attack you or even place blame` - If there's no blame, what exactly do you want?

Does the DMCA still stand? If it does, let's wait for its result. If it doesn't, cease your comments. Simple.

**Comment by pannal** [comment]
> I wanted to use your project and have people in the SC community contribute back, but currently I am unable to do that with controversy and lack of trust.

Then don't.

**Comment by WebSpider** [comment]
@pannal Now that you ask:

The MIT license at https://github.com/Anorov/cloudflare-scrape/blob/master/LICENSE requires contribution to Anorov.

This was done in commit #1 in this repo, and has not yet been repaired 

This repo has lost the original copyright claim, and thus infringes on the original authors rights.

With MIT license it should be pretty simple: Restore proper attributions in the LICENSE file and this should be fixed. MIT doesn't require the entire commit history (although it seems fair to the original authors to include it)

I'm all for using an actively developed cfscrape, but given the copyright nonsense we've had in the SickChill project, I think this should be repaired in this repo, before we can switch to this component from Anorovs.

If you like, I can PR all the attributions I think are needed to fix this.

**Comment by miigotu** [comment]
Not sure who @pannal even is, so there is nothing to discuss with them. I don't need a welcome, I've been doing this likely longer than you have been alive.
You can want something without attacking someone. I want this project to give the credit it owes to its previous contributors and come back into compliance with community standards so that we can embrace the project and use it, otherwise its just a pile of files in the corner.

If the adults want to talk about this, you know how to reach me.

**Comment by pannal** [comment]
@miigotu yeah sorry for the harsh response. I was a bit in over my head. I'm not directly involved with this project, so just ignore my comments :)

**Comment by WebSpider** [comment]
@miigotu it appears @VeNoMouS fixed the attribution in ead8f19

**Comment by VeNoMouS** [comment]
Can we all please act civil and not feed off each others hostility, I have chosen not to engage in this bullshit for a reason. 

@miigotu I did send you an email, along time ago (05/08)
